File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    min_num = 1
    max_num = 1
    _sum = 0

    while min_num <= n:
        if _sum < n:
            _sum += max_num
            max_num += 1
        elif _sum > n:
            _sum -= min_num
            min_num += 1
        else:
            answer += 1
            _sum -= min_num
            min_num += 1

    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        System.out.println();
        List<Integer> list = new ArrayList<>();
        
        list.add((int)n/2);
        list.add((int)n/2 +1);
        
        int max = (int) n/2 + 1;
        int min = (int) n/2;
        
        // 15
        int now = max + min;
        
        while (min >= 1) {
            // System.out.println("now: "+now);
            // System.out.println("max: "+max);
            // System.out.println("min: "+min);
            if (now == n) {
                answer += 1;
                now = now - max;
                max = max -1;
            }
            // 합계가 n보다 크면
            else if (now > n) {
                now = now - max;
                max = max -1;
            }
            // 합계가 n 보다 작을떄
            else if (now < n) {
                min = min -1;
                now = now + min;
            }
            
        }
        
        return answer+1;
    }
}